syntax = "proto3";

package product;

service ProductService {
    rpc GetNewProducts(Empty) returns (Products);
    rpc MarkProductSaved(Product) returns (Empty);
    rpc GetSavedProducts(Empty) returns (Products);
    rpc DebugService(MergeRequest) returns (Empty);
}

message Empty {}

message Product {
    string id = 1;
    string name = 2;
    string description = 3;
    double price = 4;
}

message Products {
    repeated Product products = 1;
}

message MergeRequest {
    map<string, InputValue> input = 1;
}

message InputValue {
    string string_value = 1;
    double double_value = 2;
    InputValue nested_value = 3;
}
